### PR TITLE
Add sortBy Navigation Feature

### PR DESCRIPTION
**Pull request by sitenzo**
## Summary
This pull request introduces a new sorting feature in the Filament Quick Create package. The update allows sorting of items by their Resource navigation sort order using `->sortBy('navigation')`. This enhancement improves consistency and user experience by aligning items according to the Filament Resource navigation order.

## Key Changes
- **SortBy Navigation Implementation**: Integrated the `->sortBy('navigation')` method, enabling resources to be sorted based on their defined navigation sort order in the Filament admin panel.
- **Code Refinement**: Ensured compatibility with existing functionalities and conducted comprehensive testing for stability and performance.
- **Documentation Update**: Revised the README and other relevant documentation to include guidance and examples for the new sorting feature.

## Impact
- **Enhanced User Experience**: Facilitates a more intuitive and streamlined user interface by displaying resources in an order that reflects their setup in the Filament admin panel.

## Additional Notes
- Open to feedback and suggestions for further enhancements or improvements.

### CODE REVIEW

File: src/QuickCreatePlugin.php [hunk 1]
```diff
@@ -150,4 +151,15 @@ public function sort(bool | Closure $condition = true): static
 
         return $this;
     }
+
+    public function sortBy(string | Closure $sortBy = 'label'): static
+    {
+        if(!in_array(['label','navigation'])){
```
Comment (awcodes) [inline src/QuickCreatePlugin.php]: You're missing the 'needle' for in_array().

```php
if(! in_array($sortBy, ['label','navigation'])){
    $sortBy = 'label';
}
```

File: README.md [hunk 2]
```diff
@@ -101,6 +101,23 @@ public function panel(Panel $panel): Panel
 }
 ```
 
+### Sorting by recourse navigation
```
Comment (awcodes) [inline README.md]: Should be 'resource' not 'recourse'

File: README.md [hunk 3]
```diff
@@ -101,6 +101,23 @@ public function panel(Panel $panel): Panel
...
+
+By default, Quick Create will sort all the displayed options by Label. This can be changed to resource navigation sort should you choose. In which case they will be displayed in the order they are displayed in de navigation.
```
Comment (awcodes) [inline README.md]: Please fix 'de' to be 'the'

**Review by awcodes — CHANGES_REQUESTED** [review]
You missing the $sortBy property on the class. It will throw an error if ->sortBy() isn't used when registering the plugin.

```php
protected string | Closure $sortBy = 'label';
```

**Comment by sitenzo** [comment]
Hi @awcodes,
Thanks for the feedback.

**Review by awcodes — APPROVED** [review]

**Pull request merged**